async_grpc/client.h

```diff
@@ -36,6 +37,17 @@ class Client {
   using ResponseType = typename RpcServiceMethod::ResponseType;
 
  public:
+  Client(std::shared_ptr<::grpc::Channel> channel, common::Duration timeout,
+         RetryStrategy retry_strategy)
+      : Client(channel, retry_strategy) {
+    timeout_ = timeout;
+  }
+
+  Client(std::shared_ptr<::grpc::Channel> channel, common::Duration timeout)
+      : Client(channel) {
+    timeout_ = timeout;
```

Comment:
Why not have an `optional<common::Duration> timeout = absl::nullopt` as default parameter for the constructors. Then you wouldn't have to introduce 2 more constructors. Also, I think it should be possible to initialize `timeout_` in constructor initialization list.